pycyphal/transport/can/media/socketcand/_socketcand.py

```diff
@@ -0,0 +1,356 @@
+
+"""
+Note: This Media interface functions almost indentically to PythonCANMedia, the only reason
+    for a completely different Media type is due to extra two variables, host and port, that none
+    of the other PythonCAN compatible interfaces take. In the future, hopefully this Media type
+    will be implimented without the use of PythonCAN at all.
+
+"""
+
+from __future__ import annotations
+import queue
+import time
+import typing
+import asyncio
+import logging
+import threading
+from functools import partial
+import dataclasses
+import warnings
+
+import can
+from pycyphal.transport import Timestamp, ResourceClosedError, InvalidMediaConfigurationError
+from pycyphal.transport.can.media import Media, FilterConfiguration, Envelope, FrameFormat, DataFrame
+
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass(frozen=True)
+class _TxItem:
+    msg: can.Message
+    timeout: float
+    future: asyncio.Future[None]
+    loop: asyncio.AbstractEventLoop
+
+
+@dataclasses.dataclass(frozen=True)
+class PythonCANBusOptions:
+    hardware_loopback: bool = False
+    """
+    Hardware loopback support.
+        If True, loopback is handled by the supported hardware.
+        If False, loopback is emulated with software.
+    """
+    hardware_timestamp: bool = False
+    """
+    Hardware timestamp support.
+        If True, timestamp returned by the hardware is used.
+        If False, approximate timestamp is captured by software.
+    """
+
+
+class SocketcandMedia(Media):
+    # pylint: disable=line-too-long
+    """
+    Media interface adapter for `Socketcand <https://github.com/linux-can/socketcand/tree/master>` using the
+    built in interface compatibility from `Python-CAN <https://python-can.readthedocs.io/>`.
+    Please refer to the Socketcand documentation and the Python-CAN documentation for information about
+    supported hardware, configuration, and installation instructions.
+
+    This media interface supports both Classic CAN and CAN FD. The selection logic is documented below.
+
+    Here is a basic usage example based on the Yakut CLI tool.
+    Suppose you have two computers:
+    One connected to a CAN capable device and that computer is able to connect and recieve CAN data from the
+    CAN device. Using socketcand with a command such as `socketcand -v -i can0 -l 123.123.1.123`
+    on this first computer will bind it too a socket (default port for socketcand is 29536, so it is also default here).
+
+    On your second computer:
+
+        export UAVCAN__CAN__IFACE="socketcand:can0:123.123.1.123"
+        export UAVCAN__CAN__BITRATE='500000'
+        export UAVCAN__CAN__MTU=8
+        yakut sub 33:uavcan.si.unit.voltage.scalar
+
+    This will allow you to wirelessly recieve CAN data on computer two through the wired connection on computer 1.
+
+    """
+
+    _MAXIMAL_TIMEOUT_SEC = 0.1
+
+    def __init__(
+        self,
+        iface_name: str,
```

Comment:
As this is not a unified adapter on top of PythonCAN, there is no point in accepting a baked spec string instead of taking the required parameters directly. Please replace this with separate channel, host, and port parameters.

Reply:
How would you suggest I update the _transport_factory file? I used the baked spec string implementation because it seemed to make the most sense given the _make_can method in the transport factory file.

Reply:
I would parse the spec string in `_make_can` and then pass the decoded values to the constructor here. Actually, we should be doing the same for `PythonCANMedia` as well, but that is a separate story...